Cabal/Distribution/Package.hs

```diff
@@ -18,7 +18,7 @@
 
 module Distribution.Package (
         -- * Package ids
-        PackageName(..),
+        PackageName, unPackageName, mkPackageName,
```

Comment:
It is too bad we didn't have the foresight to add these functions ahead of time. This means that it will be difficult to be backwards and forwards compatible in a Custom setup.